src/auth/client/mod.rs

```diff
@@ -78,48 +80,159 @@ pub enum LoginResponse {
 		redirect_address: (LuString33, u16),
 		/// Whether the account is in free trial mode.
 		is_ftp: bool,
+		/// Whether the account is connecting as a paid account for the first time.
+		just_upgraded_from_ftp: bool,
+		language: Language,
+		country_code: LuString3,
+		auth_logs: Vec<LogEntry>,
```

Comment:
I forgot to mention this, but the client calls these "stamps", would be good to keep that name for consistency.